pvlib/iotools/sodapro.py

```diff
@@ -0,0 +1,368 @@
+        service and the error message will be raised as an exception.
+
+    References
+    ----------
+    .. [1] `CAMS Radiation Service Info
+       <http://www.soda-pro.com/web-services/radiation/cams-radiation-service/info>`_
+    .. [2] `CAMS McClear Service Info
+       <http://www.soda-pro.com/web-services/radiation/cams-mcclear/info>`_
+    .. [3] `CAMS McClear Automatic Access
+       <http://www.soda-pro.com/help/cams-services/cams-mcclear-service/automatic-access>`_
+    """
+    try:
+        time_step_str = TIME_STEPS_MAP[time_step]
+    except KeyError:
+        raise ValueError(f'Time step not recognized. Must be one of '
+                         f'{list(TIME_STEPS_MAP.keys())}')
+
+    if (verbose is True) & ((time_step != '1min') | (time_ref != 'UT')):
+        verbose = False
+        warnings.warn("Verbose mode only supports 1 min. UT time series!")
+
+    if identifier not in ['mcclear', 'cams_radiation']:
+        raise ValueError('Identifier must be either mcclear or cams_radiation')
+
+    # Format verbose variable to the required format: {'true', 'false'}
+    verbose = str(verbose).lower()
+
+    if altitude is None:  # Let SoDa get elevation from the NASA SRTM database
+        altitude = -999
+
+    # Start and end date should be in the format: yyyy-mm-dd
+    start_date = start_date.strftime('%Y-%m-%d')
+    end_date = end_date.strftime('%Y-%m-%d')
+
+    email = email.replace('@', '%2540')  # Format email address
+    identifier = 'get_{}'.format(identifier.lower())  # Format identifier str
+
+    base_url = f"http://{server}/service/wps"
+
+    data_inputs = (
+        f"latitude={latitude};longitude={longitude};altitude={altitude};"
+        f"date_begin={start_date};date_end={end_date};time_ref={time_ref};"
+        f"summarization={time_step_str};username={email};verbose={verbose}")
+
+    params = {'Service': 'WPS',
+              'Request': 'Execute',
+              'Identifier': identifier,
+              'version': '1.0.0',
+              'RawDataOutput': 'irradiation',
+              }
+
+    # The DataInputs parameter of the URL has to be manually formatted and
+    # added to the base URL as it contains sub-parameters seperated by
+    # semi-colons, which gets incorrectly formatted by the requests function
+    # if passed using the params argument.
+    res = requests.get(base_url + '?DataInputs=' + data_inputs, params=params,
+                       timeout=timeout)
+
+    # Invalid requests returns an XML error message and the HTTP staus code 200
+    # as if the request was successful. Therefore, errors cannot be handled
+    # automatic (e.g. res.raise_for_status()) and errors are handled manually
+    if res.headers['Content-Type'] == 'application/xml':
+        errors = res.text.split('ows:ExceptionText')[1][1:-2]
+        raise requests.HTTPError(errors, response=res)
+    # Successful requests returns a csv data file
+    elif res.headers['Content-Type'] == 'application/csv':
```

Comment:
Should we be safe and add an `else` to this `if/elif` chain?  Or is that in [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) territory

Reply:
@wholmgren have previously recommended against this. See this previous [discussion](https://github.com/pvlib/pvlib-python/pull/1175#discussion_r584194688).